### PR TITLE
NEW: Add CUDA builds and update dep pinnings

### PR DESCRIPTION
**Pull request by carterbox**
Enables CUDA build targets and removes strict pytorch version pinning.

NOTE: flashattention only supports archs 80 and later, but it seems there are other parts of xformers that don't use flashattention, so I still request to build for all archs compatible with the CUDA compiler.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by carterbox** [comment]
@conda-forge-admin, please rerender

**Comment by carterbox** [comment]
@conda-forge-admin, please rerender

**Comment by carterbox** [comment]
I think the CUDA jobs are stalling/becoming unresponsive because RAM usage is too high. On my personal machine, the CUDA builds report a maximum memory use of 8.6GB, which is larger than the [7GB available to free Azure pipelines runners.](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#hardware)

**Comment by carterbox** [comment]
@conda-forge-admin, please rerender

**Comment by shermansiu** [comment]
@jan-janssen Could you please review this, when you have the time?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -6,32 +6,45 @@ package:
...
+  skip: true  # [cuda_compiler_version == "11.0"]
+  skip: true  # [cuda_compiler_version == "11.1"]
+  rpaths:
+    - lib/
     # conda-forge::pytorch provides these libs
-    - $RPATH/libtorch_cpu.so
-    - $RPATH/libtorch_cpu.dylib
-    - $RPATH/libc10.so
-    - $RPATH/libc10.dylib
+    - {{ SP_DIR }}/torch/lib/
```
Comment (jan-janssen) [inline recipe/meta.yaml:29]:
```suggestion
   - {{ SP_DIR }}/torch/lib/
  # weigh down gpu implementation and give nocuda preference
  track_features:
    - cudatoolkit               # [cuda_compiler_version != "None"]
 ```

**Comment by carterbox** [comment]
@conda-forge-admin, please rerender

**Comment by carterbox** [comment]
@conda-forge-admin, please rerender

**Review by jan-janssen — APPROVED** [review]
Looks good to me

**Comment by jan-janssen** [comment]
@carterbox Do you want to join me in maintaining this package? If yes, feel free to add yourself to the maintainers in the `meta.yaml` file and do not forget to rerender it after doing so.

**Comment by carterbox** [comment]
No, thanks!

**Pull request merged**